### PR TITLE
2021q3 release

### LINKED ISSUE #646: Upgrade CMSIS-DSP to v5.8.0
CMSIS-DSP has changed quite a bit:

- the API has been split up into separate headers with `arm_math.h` just including them all. This needs to be generated for the modularity.
- has added support for f16 floating point format, which isn't natively supported by any FPU operations, however, there is an instruction for converting f16 to/from f32.  Should be filtered out if not applicable to ARMv7-M
- Better support for Cortex-M0 or Hosted. We should support that.
- Better config management via `<arm_math_conf_local.h>`, for this the header `arm_math_types.h` is included by every source and header, so it should be patched to include that.
- Update examples and test them.

### PR DESCRIPTION
**Pull request by salkinium**
It's quarterly release time again! 2021-10-01 is the date.

- [x] Merge remaining PRs:
   - [x] ADC F3 fixes
   - [x] GPIO rework
- [x] Update all submodules
- [x] Fix submodule issues: None so far…
- [x] Update release notes
- [x] Check for updated compilers
- [x] Rework documentation to point to Discussions instead of Issues for questions/ideas/projects.
- [x] What about CMSIS 5.8.0 ???
- [x] Hardware testing
   - [x] TinyUSB: No breaking changes for STM32
   - [x] CMSIS-DSP: No breaking changes, added two new examples

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,6 +52,158 @@ pay attention to. Medium impact changes are also worth looking at.
...
+- LVGL upgraded to v8.0.2.
+- TinyUSB upgraded to v0.11.0.
+- FreeRTOS upgraded to 202107.
+- CMSIS-DSP upgraded to v5.8.0.
+
+Fixes:
+
+- Longer modm::delay_ms implementation.
+- Lower C++ binary size due to not using `atexit` function.
+- SPI mode 2/3 was swapped in interface, mode 1/2 swapped on AVR.
```
Comment (rleh) [inline CHANGELOG.md]:
```suggestion
- SPI mode 3/4 was swapped in interface, mode 1/2 swapped on AVR.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -52,6 +52,158 @@ pay attention to. Medium impact changes are also worth looking at.
...
+- Enable FPU in assembly before startup script.
+- Fix inconsistent flags API for STM32F3 ADC.
+- Refactored GpioConnector to remove `template< Peripheral _ >` signal
+  boilerplate.
+
+New development boards:
```
Comment (rleh) [inline CHANGELOG.md:97]: + DevEBox STM32H750VB

File: CHANGELOG.md [hunk 3]
```diff
@@ -52,6 +52,158 @@ pay attention to. Medium impact changes are also worth looking at.
...
+
+PR [#679][] -> [93bba13][].  
+Tested in hardware by [@mcbridejc][].
+
+#### 2021-09-09: Add SAMG55 support
+
+Also adds NUCLEO-H743ZI and NUCLEO-H723ZG boards and examples.
+
+PR [#673][] -> [2c22fae][].  
+Tested in hardware by [@mcbridejc][].
```
Comment (rleh) [inline CHANGELOG.md]:
```suggestion
#### 2021-09-09: Add SAMG55 support

Also adds SAMG55-XPLAINED-PRO board and blink example.

PR [#676][] -> [c148bf8][].  
Tested in hardware by [@mcbridejc][].
```

File: docs/release/2021q3.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+PR #679 -> 93bba13.  
+Tested in hardware by @mcbridejc.
+
+#### 2021-09-09: Add SAMG55 support
+
+Also adds NUCLEO-H743ZI and NUCLEO-H723ZG boards and examples.
+
+PR #673 -> 2c22fae.  
+Tested in hardware by @mcbridejc.
```
Comment (rleh) [inline docs/release/2021q3.md]: see above...

File: CHANGELOG.md (hunk 3 above)
Comment (salkinium) [reply CHANGELOG.md]: 🙈 too much copy-paste for me…

**Pull request merged**